claat/render/md.go

```diff
@@ -323,21 +324,46 @@ func (mw *mdWriter) youtube(n *types.YouTubeNode) {
 }
 
 func (mw *mdWriter) table(n *types.GridNode) {
+	// Calculate table inner content length.
+	contentLength := 0
+	for _, row := range n.Rows {
+		for _, cell := range row {
+			var nw bytes.Buffer
+			WriteMD(&nw, mw.env, mw.format, cell.Content.Nodes...)
+			contentLength += len(bytes.Trim(nw.Bytes(), " \t"))
+		}
+	}
+	// If table content is empty, don't output the table.
+	if contentLength == 0 {
+		return
+	}
+
 	mw.writeBytes(newLine)
 	maxcols := maxColsInTable(n)
 	for rowIndex, row := range n.Rows {
 		mw.writeString("|")
 		for _, cell := range row {
 			mw.isWritingTableCell = true
 			mw.writeString(" ")
-			for _, cn := range cell.Content.Nodes {
-				cn.MutateBlock(false) // don't treat content as a new block
-				mw.write(cn)
+
+			// Check cell content for newlines and replace with inline HTML if newlines are present.
+			var nw bytes.Buffer
+			WriteMD(&nw, mw.env, mw.format, cell.Content.Nodes...)
+			if bytes.ContainsRune(nw.Bytes(), '\n') {
+				for _, cn := range cell.Content.Nodes {
+					cn.MutateBlock(false) // don't treat content as a new block
+					var nw2 bytes.Buffer
+					WriteHTML(&nw2, mw.env, mw.format, cn)
+					mw.writeBytes(bytes.ReplaceAll(nw2.Bytes(), []byte("\n"), []byte("")))
```

Comment:
Consider replacing new lines with `<br>` or ` ` rather than just removing them. <br> is probably the best path forward.

Reply:
I think in this case it's newlines in the markdown source that we're replacing, so we probably don't want <br />.

e.g. we are changing this:
<ul>
  <li>thing</li>
</ul>

to this:
<ul><li>thing</li></ul>

If the goal is to preserve newlines in text, we should probably add something to the text render method that replaces them with <br />, but leave this as-is.

Reply:
Shawn is right, replacing `\n` with `<br>` would be better placed in text rendering logic.

Reply:
I think we can do that in another PR - but Shawn should address the other comment above before merging.